pkg/container-utils/testutils/containerd.go

```diff
@@ -34,34 +35,71 @@ const (
 	defaultNamespace = "k8s.io"
 )
 
-func RunContainerdContainer(ctx context.Context, t *testing.T, command string, options ...Option) {
-	opts := defaultContainerOptions()
+func NewContainerdContainer(name, cmd string, options ...Option) Container {
+	c := &ContainerdContainer{
+		containerSpec: containerSpec{
+			name:    name,
+			cmd:     cmd,
+			options: defaultContainerOptions(),
+		},
+	}
 	for _, o := range options {
-		o(opts)
+		o(c.options)
 	}
+	return c
+}
 
-	nsCtx := namespaces.WithNamespace(ctx, defaultNamespace)
-	fullImage := getFullImage(opts)
+type ContainerdContainer struct {
+	containerSpec
+
+	client     *containerd.Client
+	nsCtx      context.Context
+	exitStatus <-chan containerd.ExitStatus
+}
 
-	client, err := containerd.New("/run/containerd/containerd.sock",
+func (c *ContainerdContainer) Running() bool {
+	return c.started
+}
+
+func (c *ContainerdContainer) ID() string {
+	return c.id
+}
+
+func (c *ContainerdContainer) Pid() int {
+	return c.pid
+}
+
+func (c *ContainerdContainer) initClientAndCtx() error {
+	var err error
+	c.client, err = containerd.New("/run/containerd/containerd.sock",
```

Comment:
Use the default value from the following file? https://github.com/inspektor-gadget/inspektor-gadget/blob/03c29258d42e7d13126b7a83da8f89ce4e8d030e/pkg/container-utils/runtime-client/interface.go#L29

Reply:
I see.
Then I would additionally create a very simple and small unittest which checks if the default socket path is still the one you set here manually. If it changes we can see the failed test directly and know what happened.

If we don't have such a test and a hardcoded value all tests (might) fail and we have to "search" for the reason.

WDYT?

Reply:
IMO, the output is clear enough: 

```
=== CONT  TestRuntimeClientInterface/containerd
    containerd.go:88: Failed to initialize client: creating a client: failed to dial "/invalid/path/to/containerd.sock": context deadline exceeded: connection error: desc = "transport: error while dialing: dial unix:///invalid/path/to/containerd.sock: timeout"
```

WDYT?

Reply:
Seems good enough :+1: